smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/commit/KafkaThrottledLatestProcessedCommit.java

```diff
@@ -158,27 +160,37 @@ private void startFlushAndCheckHealthTimer() {
 
     }
 
-    private void flushAndCheckHealth(long timerId) {
-        Map<TopicPartition, Long> offsetsMapping = clearLesserSequentiallyProcessedOffsetsAndReturnLargestOffsetMapping();
-
-        if (!offsetsMapping.isEmpty()) {
-            Map<TopicPartition, OffsetAndMetadata> offsets = offsetsMapping
-                    .entrySet()
-                    .stream()
-                    .collect(Collectors.toMap(Map.Entry::getKey,
-                            e -> new OffsetAndMetadata().setOffset(e.getValue() + 1L)));
-            consumer.getDelegate().commit(offsets, a -> this.startFlushAndCheckHealthTimer());
-        } else {
-            this.startFlushAndCheckHealthTimer();
+    private synchronized Context getContext() {
+        if (context == null) {
+            context = vertx.getOrCreateContext();
         }
+        return context;
+    }
```

Comment:
I ran a test locally on a project and it works perfectly!

I do have one final suggestion. **context** here is loaded only once, so using a lock every time we acquire seems heavy handed. What do you think about using a double checked lock? We would have to change context to be volatile on line 51 

```
private volatile Context context;
```

And then rewrite this method

```
    private Context getContext() {
        Context ctx = this.context;
        if (ctx == null) {
            synchronized (this) {
                ctx = this.context;
                if (ctx == null) {
                    this.context = ctx = vertx.getOrCreateContext();
                }
            }
        }
        return ctx;
    }
```

We could do the same with the _latest_ strategy - maybe even abstract it to reduce duplicate code.

Theoretically we would see a 100x performance gain.

Reply:
Fixed.